smart-contracts/test/Lock/purchaseGasRefund.js

```diff
@@ -0,0 +1,143 @@
+const truffleAssert = require('truffle-assertions')
+const BigNumber = require('bignumber.js')
+const deployLocks = require('../helpers/deployLocks')
+const getProxy = require('../helpers/proxy')
+
+const unlockContract = artifacts.require('Unlock.sol')
+
+let unlock
+let locks
+const keyPrice = web3.utils.toWei('0.01', 'ether')
+
+contract('Lock / GasRefund', (accounts) => {
+  let lock
+  const tokenAddress = web3.utils.padLeft(0, 40)
+  let userETHBalanceBefore
+
+  beforeEach(async () => {
+    unlock = await getProxy(unlockContract)
+    locks = await deployLocks(unlock, accounts[0], tokenAddress)
+    lock = locks.FIRST
+  })
+
+  describe('gas refund percentage', () => {
+    it('get set properly', async () => {
+      await lock.setGasRefundPercentage(25)
+      assert.equal(await lock.gasRefundPercentage(), 25)
+    })
+
+    it('can not be set if caller is not lock manager or beneficiary', async () => {
+      await truffleAssert.fails(
+        lock.setGasRefundPercentage(25, {
+          from: accounts[3],
+        }),
+        'revert',
+        'ONLY_LOCK_MANAGER_OR_BENEFICIARY'
+      )
+    })
+
+    it('can be by set beneficiary', async () => {
+      await lock.setGasRefundPercentage(25, { from: accounts[0] })
+      assert.equal(await lock.gasRefundPercentage(), 25)
+    })
+
+    it('can set by lock manager', async () => {
+      await lock.addLockManager(accounts[5], { from: accounts[0] })
+      await lock.setGasRefundPercentage(25, { from: accounts[5] })
+      assert.equal(await lock.gasRefundPercentage(), 25)
+    })
+  })
+
+  describe('purchase with gas refund', () => {
+    let tx
+
+    beforeEach(async () => {
```

Comment:
Let's also write tests for an ERC20 just to make sure :)

Reply:
yep added